middlewares/replace_path_test.go

```diff
@@ -0,0 +1,44 @@
+package middlewares_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/containous/traefik/middlewares"
+)
+
+func TestReplacePath(t *testing.T) {
+	const replacementPath = "/replacement-path"
+
+	var newPath, oldPath string
+	handler := &middlewares.ReplacePath{
+		Path: replacementPath,
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			newPath = r.URL.Path
+			oldPath = r.Header.Get("X-Replaced-Path")
+		}),
+	}
```

Comment:
Please move the block from line 13 to 20 into the test loop body in order to make sure the tests work okay even if we run them in parallel.